00-RELEASENOTES

```diff
@@ -1,16 +1,162 @@
+* Automatic failover vote is not limited by two times the node timeout (#1356)
+
+Behavior Changes
+================
+* Streams use an additional 8 bytes to track their internal size (#688)
+* Take hz into account in activerehashing to avoid CPU spikes (#977)
+* Incr expired_keys if the expiration time is already expired (#1517)
+* Fix replica not able to initiate election in time when epoch fails (#1009)
+* Make FUNCTION RESTORE FLUSH flush async based on lazyfree-lazy-user-flush (#1254)
+* Allow `MEMORY MALLOC-STATS` and `MEMORY PURGE` during loading phase (#1317)
+* Use `DEEPBIND` flag when loading external modules in order to avoid symbol conflicts (#1703)
+
+
+Logging and Tooling Improvements
+================================
+* Configurable log and timestamp formats (logfmt, ISO8601) (#1022)
+* Remove the restriction that cli --cluster create requires at least 3 primary nodes (#1075)
+* Add short client info log to CLUSTER MEET / FORGET / RESET commands (#1249)
+* Support for reading from replicas in valkey-benchmark (#1392)
+* valkey-cli: re-select previously selected database after reconnect (#1694)
+* valkey-cli auto-exit from subscribed mode (#1432)
+
+Bug Fixes
+=========
+* Fix module / script call CLUSTER SLOTS / SHARDS fake client check crash (#1063)
+* Use _Thread_local to solve threads.h build issue (#1053)
+* Fix RDMA build dependency (#1074)
+* Fix primary crash when processing dirty slots during shutdown wait / failover wait / client pause (#1131)
+* Fix empty response for ACL CAT category subcommand for module defined categories (#1140)
+* Reset repl_down_since to zero only on state change (#1149)
+* Add 'WithDictIndex' expiry API and update RANDOMKEY command (#1155)
+* Fix FUNCTION KILL error message being displayed as SCRIPT KILL (#1171)
+* Add scoped RDB loading context and immediate abort flag (#1173)
+* Fix SORT GET to ignore special pattern # in cluster slot check (#1182)
+* Mark the node as FAIL when the node is marked as NOADDR and broadcast the FAIL (#1191)
+* [Bug Fix] Optimize RDB Load Performance and Fix Cluster Mode Resizing (#1199)
+* RDMA: Delete keepalive timer on closing (#1237)
+* Fix incorrect cache_memory reset in functionsLibCtxClear (#1255)
+* Log as primary role (M) instead of child process (C) during startup (#1282)
+* Fix empty primary may have dirty slots data due to bad migration (#1285)
+* Fix LRU crash when getting too many random lua scripts (#1310)
+* RDMA: Fix dead loop when transfer large data (20KB) (#1386)
+* Fix: restore replica output bytes stat update (#1486)
+* Use the correct command proc for the LOOKUP_NOTOUCH exception in lookupKey (#1499)
+* Fix for when active rehashing kvstore dictionaries are replaced by defrag (#1512)
+* Fix module LatencyAddSample still work when latency-monitor-threshold is 0 (#1541)
+* Fix default value of primary-reboot-down-after-period in sentinel.conf (#1040)
+
+We appreciate the efforts of all who contributed code to this release!
```

Comment:
Is this based on the code contribution ranking of version 8.1? What tool was used?

Reply:
@soloestoy I just used github cli to get all the PR contributors of merged PRs which were merged AFTER 8.0.0 - is there another preferred way?

Reply:
I'm just curious : )

Reply:
something in the form of:
```
gh pr list --limit 100000 --state merged --json mergedAt,author --jq '.[] | select(.mergedAt > "'"$(git show -s --format=%ci $TAG_COMMIT_HASH)"'") | "\(.author.name)(\(.author.login))"' | sort -u | paste -sd "," -
```

Reply:
:medal_sports: Let's add this to the wiki page about how to make releases.

Reply:
>Let's add this to the wiki page about how to make releases. 

Oh after I am done with this release I will make sure to document all the steps. For example it was not clear that we need to create a branch (8.1) first, and the steps to create it are not trivial. also the way I gather the commits was:

```
gh pr list --limit 100000 --state merged --label "release-notes" --json number,mergedAt,title   --jq '.[] | select(.mergedAt > "'"$(git show -s --format=%ci $TAG_COMMIT_HASH)"'") | "\(.title) (#\(.number))"'
```